File: Team2/Riya_Gupta/MonotonicArray.cpp
class Solution {
public:
    bool isMonotonic(vector<int>& A) {
        
        
        int length = A.size() ; 
        
        if(length == 0 || length == 1)
            return true;
        
        if(A[0] <= A[length-1])
        {
            for(int i = 0 ; i < length -1 ; i++)
            {
                if(A[i] > A[i+1])
                    return false;
                    
            }
            return true ; 
            
        }
        
        else
        {
            for(int i = 0 ; i < length -1 ; i++)
            {
                if(A[i] < A[i+1])
                    return false;
                    
            }
            return true ; 
            
        }
        
    }
};

File: Team1/Riya Tanwar/monotonic-array.java
class Solution {
    public boolean isMonotonic(int[] A) {
        int direction = -1; 
        for (int i = 0; i < A.length - 1; i++) {
            if (A[i] < A[i + 1]) {
                if (direction == -1) {
                    direction = 0;
                }
                if (direction != 0) {
                    return false;
                }
            } else if (A[i] > A[i + 1]) {
                if (direction == -1) {
                    direction = 1;
                }
                if (direction != 1) {
                    return false;
                }
            }
        }
        return true;
    }
}